Plugin data fetchers

keller-mark opened the pull request:
## Description

> What was changed in this pull request?

This PR implements plugin data fetcher functionality by looking up a `window.higlassDataFetchersByType` object.

> Why is it necessary?

I have implemented a Zarr-based "serverless" multivec data fetcher class, but this requires the zarr.js dependency. Instead of adding this dependency to HiGlass i would like to add it with a "plugin datafetcher". 

I moved `minNonZero` and `maxNonZero` out of `./worker` and into the `./utils/` directory since utils are exposed in `AVAILABLE_TO_PLUGINS`

Fixes #___

## Checklist

- [ ] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [x] Unit tests added or updated
- [x] Documentation added or updated
- [ ] Example(s) added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [ ] Screenshot for visual changes (e.g. new tracks or UI changes)
- [x] Updated CHANGELOG.md

pkerpedjiev commented:
This is great! I wish it existed about a week ago. 

I'd love to try it out if you have a template repo I could use. In fact, if this works then maybe https://github.com/higlass/higlass/pull/964 could be implemented as a plugin data fetcher?

Add some docs and it should be good to go 👍

keller-mark commented:
This is an example repo for the zarr-based multivec plugin datafetcher https://github.com/keller-mark/higlass-zarr-multivec-datafetcher

keller-mark commented:
I added a documentation page in this commit https://github.com/higlass/higlass/pull/966/commits/680858f03b14397fcfe31e6877260976fbc3c783 and moved the list of internals available to plugins to a common page since they will be available to both tracks and data fetchers.
